### PR TITLE
fix: workload promote test and status code

### PR DESCRIPTION
**Pull request by dimakis**
## Summary

- Fix vacuous broadcast test: `setWorkloadBroadcast()` instead of monkey-patching non-existent `app._workloadBroadcast`
- Add missing test for promote with description but no title
- Return 400 (not 404) when item exists but has no title

Pre-existing issues flagged by Centaur on PR #428.

## Test plan

- [x] workload-routes tests pass (25/25)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/__tests__/workload-routes.test.ts [hunk 1]
```diff
@@ -558,7 +567,8 @@ describe('workload routes', () => {
     const workloadBroadcasts = broadcasts.filter((b: any) => b.type === 'workload_item_updated');
     expect(workloadBroadcasts).toHaveLength(0);
 
-    (app as any)._workloadBroadcast = origBroadcast;
+    // Reset — no broadcast callback in test environment
```
Comment (dimakis) [inline server/__tests__/workload-routes.test.ts:570]: 🔵 **style**: The reset comment '// Reset — no broadcast callback in test environment' explains what the line does, not why. The code `setWorkloadBroadcast(() => {})` is self-explanatory. Per project conventions, this comment should be removed. `[fixable]`

**Review by dimakis — COMMENTED** [review]
## Centaur Review

Found **2** issue(s) (1 warning).


### `server/__tests__/workload-routes.test.ts`
_Good fixes — proper use of setWorkloadBroadcast API and correct 400/404 status differentiation — but the new 400 branch (item exists, no title) lacks test coverage._

- 🟡 **missing_tests**: The code change introduces a new 400 status path (item exists in workloadStore but has no title), but there's no test covering it. The new test only covers the 404 path (missing item). Add a test that creates an item with an empty-string title, then promotes it without providing a body title, and asserts 400 with 'No title provided'. `[fixable]`
- 🔵 **style** (L570): The reset comment '// Reset — no broadcast callback in test environment' explains what the line does, not why. The code `setWorkloadBroadcast(() => {})` is self-explanatory. Per project conventions, this comment should be removed. `[fixable]`
